presto-orc/src/main/java/com/facebook/presto/orc/writer/SegmentedSliceBlockBuilder.java

```diff
@@ -0,0 +1,416 @@
+        offsets = new int[initialSize][];
+        closedSlices = new Slice[initialSize];
+        offsets[0] = new int[SegmentHelper.SEGMENT_SIZE + 1];
+        openSliceOutput = new DynamicSliceOutput(expectedBytes);
+    }
+
+    public void reset()
+    {
+        openSliceOutput.reset();
+        Arrays.fill(closedSlices, null);
+        // Fill the first offset array with 0, and free up the rest of the offsets array.
+        Arrays.fill(offsets[0], 0);
+        Arrays.fill(offsets, 1, offsets.length, null);
+        openSegment = 0;
+        openSegmentOffset = 0;
+        closedSlicesRetainedSize = 0;
+        closedSlicesSizeInBytes = 0;
+    }
+
+    @Override
+    public int getPositionOffset(int position)
+    {
+        return getOffset(position);
+    }
+
+    @Override
+    public int getSliceLength(int position)
+    {
+        int offset = SegmentHelper.offset(position);
+        int segment = SegmentHelper.segment(position);
+        return offsets[segment][offset + 1] - offsets[segment][offset];
+    }
+
+    private Slice getSegmentRawSlice(int segment)
+    {
+        return segment == openSegment ? openSliceOutput.getUnderlyingSlice() : closedSlices[segment];
+    }
+
+    @Override
+    public Slice getRawSlice(int position)
+    {
+        return getSegmentRawSlice(SegmentHelper.segment(position));
+    }
+
+    @Override
+    public int getPositionCount()
+    {
+        return SegmentHelper.getPositions(openSegment, openSegmentOffset);
+    }
+
+    @Override
+    public long getSizeInBytes()
+    {
+        long arraysSizeInBytes = Integer.BYTES * (long) getPositionCount();
+        return openSliceOutput.size() + arraysSizeInBytes + closedSlicesSizeInBytes;
+    }
+
+    @Override
+    public long getRegionSizeInBytes(int positionOffset, int length)
+    {
+        throw new UnsupportedOperationException(getClass().getName());
+    }
+
+    @Override
+    public long getPositionsSizeInBytes(boolean[] positions)
+    {
+        throw new UnsupportedOperationException(getClass().getName());
+    }
+
+    private long getClosedSlicesRetainedSize()
+    {
+        return sizeOf(closedSlices) + closedSlicesRetainedSize;
+    }
+
+    private long getOffsetsRetainedSize()
+    {
+        return sizeOf(offsets) + (openSegment + 1) * sizeOf(offsets[0]);
+    }
+
+    @Override
+    public long getRetainedSizeInBytes()
+    {
+        long offsetsSize = getOffsetsRetainedSize();
+        long closedSlicesSize = getClosedSlicesRetainedSize();
+        return INSTANCE_SIZE + openSliceOutput.getRetainedSize() + offsetsSize + closedSlicesSize;
+    }
+
+    @Override
+    public void retainedBytesForEachPart(BiConsumer<Object, Long> consumer)
+    {
+        consumer.accept(openSliceOutput, openSliceOutput.getRetainedSize());
+        consumer.accept(offsets, getOffsetsRetainedSize());
+        consumer.accept(closedSlices, getClosedSlicesRetainedSize());
+        consumer.accept(this, (long) INSTANCE_SIZE);
```

Comment:
We wouldn't expect this method to be invoked I think

Reply:
Removed it.